### ISSUE TITLE
Registration process, including T&Cs

### ISSUE THREAD

**Issue by CrispinF** [exeGesIS-SDM/ELM#4]
Registration functionality, collecting key information about the user in order to assign correct role, requiring authorisation before the user gains access.
Initially the user roles are (probably):
RPA (Rural Payments Agency) staff.
ELM HEFER administrator (quasi administrative, probably key Historic England staff, perhaps ALGAO/RPA admins).

Do we need to collect any information from registering users other than name and email? (In the CS portal users are also asked for Organisation, Address, Phone, Mobile, Role applied for, historic map counties, and supporting information about role.)

**Comment by sarahpoppy** [comment]
No additional needed than name, email & supporting info about role is useful for handling the requests

**Comment by CrispinF** [comment]
@JimKerslake we'll need to configure a KVP user property for a mandatory text area with a label like:
"Please explain why you require access to the ELM HEFER portal:"
@sarahpoppy unless you'd also like a drop down of the possible roles being applied for, in which case the text box could have a different label like on CS portal. What would the roles be described as?

**Comment by sarahpoppy** [comment]
RPA registration unlikely to be required now moving to a self service model.  Question is whether land managers need to register to request a HEFER, or whether they can make the request with an online form collecting email address (twice to check for typos), and pros/cons of each approach.

I don't anticipate there will be resource in the system to validate 1000 registrations for instance.

**Comment by epennifold** [comment]
Do want land managers to register - must confirm email.

First name, last name, email and possibly postal address (@sarahpoppy to confirm).

**Comment by epennifold** [comment]
@sarahpoppy have you made a decision about what information we're collecting?

**Comment by sarahpoppy** [comment]
Yes registration process to collect name and email address only.

**Comment by CrispinF** [comment]
@joerichardsESDM to de-configure the KVP question that asks for reasons, as Sarah wants only name and email collected during registration.
Check staging and production config, as this might be outside deployment control (I can't recall!).

**Comment by CrispinF** [comment]
@joerichardsESDM I have removed the extra registration question on the staging site. It had not been deployed to production.
I also removed from appsettings.json on the develop branch, along with SMTP credentials.
So get latest.

**Comment by CrispinF** [comment]
@sarahpoppy you could perhaps now usefully register on https://hefer-data.org.uk/ and https://elm-hefer-staging.esdm.co.uk/ 
Other than the missing waffles, this could be all the info we require of users, if you approve. 
Then we could give you permissions to start filling in some of the content areas.

**Comment by sarahpoppy** [comment]
Registered okay but the wording on the email isn't informative as to why you have rec'd it.  Can you add a sentence along the lines of "Thank you for registering for the SFI HEFER portal.  Please click..."

![image](https://user-images.githubusercontent.com/25054442/116062346-b737ae80-a67b-11eb-9c96-47f7c61828ad.png)

**Comment by CrispinF** [comment]
I've overridden the default email template, adding some specific text as suggested. Will be testable after next deployment.

**Comment by epennifold** [comment]
Can confirm this has been changed
![image](https://user-images.githubusercontent.com/15983319/116225499-d56be000-a749-11eb-92df-f6bdb24b7dfb.png)

@sarahpoppy let us know if anything else to change or happy to move to 'done'?
